### PR TITLE
Analyse dumps from JoeSandbox karton. Remove unused old drakrun-prod.

### PR DESCRIPTION
**Pull request by kscieslinski**
config-extractor

### CODE REVIEW

File: karton/config_extractor/config_extractor.py [hunk 1]
```diff
@@ -55,6 +57,7 @@ class ConfigExtractor(Karton):
         },
         {"type": "analysis", "kind": "drakrun-prod"},
```
Comment (nazywam) [inline karton/config_extractor/config_extractor.py:58]: If we're removing the handler this should also be removed

File: karton/config_extractor/config_extractor.py [hunk 2]
```diff
@@ -133,10 +136,16 @@ def analyze_sample(self, sample: ResourceBase) -> None:
         else:
             self.log.info("Failed to get config")
 
-    # analyze a drakrun analysis
-    def analyze_drakrun(self, sample, path):
+    def analyze_dumps(self, sample, dumps_path, base_from_fname):
```
Comment (nazywam) [inline karton/config_extractor/config_extractor.py:139]: The `base_from_fname`  seems fine, but I'm thinking if just passing a tuple of `[filename, address, etc]` wouldn't be easier?

File: karton/config_extractor/config_extractor.py [hunk 3]
```diff
@@ -198,37 +205,72 @@ def analyze_drakrun(self, sample, path):
...
+        return int(dump.split("_")[0], 16)
+
+    def analyze_drakrun(self, sample, dumps):
+        with dumps.extract_temporary() as tmpdir:  # type: ignore
+            dumps_path = os.path.join(path, "dumps")
+            # Drakrun stores meta information in seperate file for each dump.
+            # Filter it as we want to analyse only dumps.
+            for fname in os.listdir(dumps_path):
+                if not re.match(r"^[a-f0-9]{4,16}_[a-f0-9]{16}$", fname):
+                    full_path = os.path.join(dumps_path, fname)
```
Comment (nazywam) [inline karton/config_extractor/config_extractor.py:223]: `full_path` seems to be unused

File: karton/config_extractor/config_extractor.py [hunk 4]
```diff
@@ -198,37 +205,72 @@ def analyze_drakrun(self, sample, path):
...
+                    os.remove(tmpdir + fname)
```
Comment (nazywam) [inline karton/config_extractor/config_extractor.py:224]: Is there actually a reason to remove the dump? They are not passed anywhere else anyway so it seems like a NOP right now

File: karton/config_extractor/config_extractor.py [hunk 5]
```diff
@@ -198,37 +205,72 @@ def analyze_drakrun(self, sample, path):
...
+        """
+        if "sdmp" in dump_name:
+            return int(dump_name.split(".")[3], 16)
+        elif "raw.unpack" in dump_name:
+            return int(dump_name.split(".")[4], 16)
+        elif "unpack" in dump_name:
+            return int(dump_name.split(".")[4], 16)
+
+    def analyze_joesandbox(self, sample, dumps):
+        with tempfile.TemporaryDirectory() as tmpdir:
```
Comment (nazywam) [inline karton/config_extractor/config_extractor.py:246]: I think you should be able to extract it using the `extract_temporary` method from karton.resource, or is there a reason ont to do that?

File: requirements.txt [hunk 6]
```diff
@@ -1,2 +1,2 @@
-karton.core==4.0.5
+karton-core==4.0.5
```
Comment (nazywam) [inline requirements.txt:1]:
```suggestion
karton-core==4.2.0
```

We might as well bump it

File: karton/config_extractor/config_extractor.py (hunk 4 above)
Comment (nazywam) [reply karton/config_extractor/config_extractor.py:222]: Oh, right they are iterated over in the config ripper stuff, I think I'd either handle this in the `base_from_fname` method (None returned == ignore dump) or, if we're going with the list of tuples way just ignore them there

File: karton/config_extractor/config_extractor.py (hunk 2 above)
Comment (kscieslinski) [reply karton/config_extractor/config_extractor.py:139]: Good point, I will refactor this.

File: karton/config_extractor/config_extractor.py (hunk 4 above)
Comment (kscieslinski) [reply karton/config_extractor/config_extractor.py:222]: I've decided to go with the list of tuples approach.

File: karton/config_extractor/config_extractor.py (hunk 5 above)
Comment (kscieslinski) [reply karton/config_extractor/config_extractor.py:244]: extract_temporary does not allow to specify the password

Comment (nazywam) [reply karton/config_extractor/config_extractor.py:247]: oh, right. Makes sense :+1:

**Review by nazywam — APPROVED** [review]

**Review by psrok1 — APPROVED** [review]

**Comment by kscieslinski** [comment]
I've tested this both locally using karton-playground (joesandbox side) and on internal cluster (drakrun). It seems to work fine.

**Pull request merged**